From 6589a58e4e04abed32eaeefcfa1282a9e8128f33 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Mon, 6 May 2019 07:57:40 +0200
Subject: [PATCH 03/90] add macros for clang annotations

---
 include/linux/compiler_attributes.h | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/include/linux/compiler_attributes.h b/include/linux/compiler_attributes.h
index 6b318efd8a74..de6fcbf3739c 100644
--- a/include/linux/compiler_attributes.h
+++ b/include/linux/compiler_attributes.h
@@ -253,4 +253,22 @@
  */
 #define __weak                          __attribute__((__weak__))
 
+/*
+ * clang thread safety analysis:
+ * TODO Link
+ * Only provided by clang.
+ * Naming for now: name the locking primitive, although irrelevant for analysis.
+ */
+#if __has_attribute(capability)
+# define __acquires_mutex(x)            __attribute__((acquire_capability(x)))
+# define __releases_mutex(x)            __attribute__((release_capability(x)))
+# define __try_acquires_mutex(r, x)     __attribute__((try_acquire_capability(r, x)))
+# define __requires_mutex(x)            __attribute__((requires_capability(x)))
+#else
+# define __acquires_mutex(x)
+# define __releases_mutex(x)
+# define __try_acquires_mutex(r, x)
+# define __requires_mutex(x)
+#endif
+
 #endif /* __LINUX_COMPILER_ATTRIBUTES_H */
-- 
2.17.1

