From 7b0b6c76c265840580be9282b8bd59dc3211b064 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Mon, 22 Apr 2019 18:23:40 +0200
Subject: [PATCH 36/82] fs/proc/proc_tty: annotate t_{start,stop}

---
 fs/proc/proc_tty.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fs/proc/proc_tty.c b/fs/proc/proc_tty.c
index c69ff191e5d8..b76af0c5035c 100644
--- a/fs/proc/proc_tty.c
+++ b/fs/proc/proc_tty.c
@@ -103,7 +103,7 @@ static int show_tty_driver(struct seq_file *m, void *v)
 }
 
 /* iterator */
-static void *t_start(struct seq_file *m, loff_t *pos)
+static void *t_start(struct seq_file *m, loff_t *pos) __acquires_mutex(tty_mutex)
 {
 	mutex_lock(&tty_mutex);
 	return seq_list_start(&tty_drivers, *pos);
@@ -114,7 +114,7 @@ static void *t_next(struct seq_file *m, void *v, loff_t *pos)
 	return seq_list_next(v, &tty_drivers, pos);
 }
 
-static void t_stop(struct seq_file *m, void *v)
+static void t_stop(struct seq_file *m, void *v) __releases_mutex(tty_mutex)
 {
 	mutex_unlock(&tty_mutex);
 }
-- 
2.17.1

