From 9e495ca6e965044319db846ade7fde4b47593b70 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Fri, 26 Apr 2019 04:16:08 +0200
Subject: [PATCH 59/90] include/linux/fb.h: annotate lock_fb_info

---
 include/linux/fb.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/fb.h b/include/linux/fb.h
index 16f7440298e1..aeda98350ca5 100644
--- a/include/linux/fb.h
+++ b/include/linux/fb.h
@@ -660,7 +660,7 @@ extern struct class *fb_class;
 	for (i = 0; i < FB_MAX; i++)		\
 		if (!registered_fb[i]) {} else
 
-extern int lock_fb_info(struct fb_info *info);
+extern int lock_fb_info(struct fb_info *info) __try_acquires_mutex(1, info->lock);
 
 static inline void unlock_fb_info(struct fb_info *info) __releases_mutex(&info->lock)
 {
-- 
2.17.1

