From a4556cb8cbc953983f21fd25de11c03f7b6e3290 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Sat, 29 Jun 2019 07:38:45 +0200
Subject: [PATCH 135/171] i915: annotate intel_{prepare,finish}_reset (wip)

---
 drivers/gpu/drm/i915/intel_display.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 421aac80a838..c96e42b20ef1 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -3839,7 +3839,7 @@ static bool gpu_reset_clobbers_display(struct drm_i915_private *dev_priv)
 		intel_has_gpu_reset(dev_priv));
 }
 
-void intel_prepare_reset(struct drm_i915_private *dev_priv)
+void intel_prepare_reset(struct drm_i915_private *dev_priv) __acquires_mutex(dev_priv->drm.mode_config.mutex) __no_thread_safety_analysis
 {
 	struct drm_device *dev = &dev_priv->drm;
 	struct drm_modeset_acquire_ctx *ctx = &dev_priv->reset_ctx;
@@ -3895,7 +3895,7 @@ void intel_prepare_reset(struct drm_i915_private *dev_priv)
 	state->acquire_ctx = ctx;
 }
 
-void intel_finish_reset(struct drm_i915_private *dev_priv)
+void intel_finish_reset(struct drm_i915_private *dev_priv) __releases_mutex(dev_priv->drm.mode_config.mutex) __no_thread_safety_analysis
 {
 	struct drm_device *dev = &dev_priv->drm;
 	struct drm_modeset_acquire_ctx *ctx = &dev_priv->reset_ctx;
-- 
2.17.1

